internal/providers/gitlab/repository_properties.go

```diff
@@ -68,14 +70,88 @@ func (c *gitlabClient) getPropertiesForRepo(
 		return nil, fmt.Errorf("failed to decode response: %w", err)
 	}
 
+	outProps, err := gitlabProjectToProperties(proj)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert project to properties: %w", err)
+	}
+
+	return getByProps.Merge(outProps), nil
+}
+
+func gitlabProjectToProperties(proj *gitlab.Project) (*properties.Properties, error) {
+	ns := proj.Namespace
+	if ns == nil {
+		return nil, fmt.Errorf("gitlab project %d has no namespace", proj.ID)
+	}
+	owner := ns.Path
+
+	var license string
+	if proj.License != nil {
+		license = proj.License.Name
+	}
+
 	outProps, err := properties.NewProperties(map[string]any{
+		properties.PropertyUpstreamID:     fmt.Sprintf("%d", proj.ID),
+		properties.PropertyName:           formatRepoName(owner, proj.Name),
 		properties.RepoPropertyIsPrivate:  proj.Visibility == gitlab.PrivateVisibility,
 		properties.RepoPropertyIsArchived: proj.Archived,
 		properties.RepoPropertyIsFork:     proj.ForkedFromProject != nil,
+		RepoPropertyDefaultBranch:         proj.DefaultBranch,
+		RepoPropertyNamespace:             owner,
+		RepoPropertyProjectName:           proj.Name,
+		RepoPropertyLicense:               license,
+		RepoPropertyCloneURL:              proj.HTTPURLToRepo,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to create properties: %w", err)
 	}
 
-	return getByProps.Merge(outProps), nil
```

Comment:
**question (non-blocking):** why is this merge no longer necessary?

Reply:
It's an implementation detail. It was necessary because I was forming merging what was incoming to what I'm forming here. Now, `outProps` gets populated explicitly with everything that's needed, so I no longer needed the merge. It's just a little more efficient to not need to iterate over the incoming properties.